File: HackerRank/JavaScript/Grid Challenge.js
'use strict';

/**
 * @param {number[]} grid - Array of profit values
 * @returns {number} - Number of pairs that sum to target
 */
function gridChallenge(grid) {
    // Write your solution here

    let sorted_grid = []
    for (let row = 0; row < grid.length; row++) {
        sorted_grid.push(grid[row].split('').sort())
    }

    for (let col = 0; col < sorted_grid[0].length; col++) {
        for (let row = 1; row < sorted_grid.length; row++) {
            if (sorted_grid[row - 1][col] > sorted_grid[row][col]) {
                return "NO"
            }
        }
        return "YES"
    }
}
const grid1 = ["ebacd", "fghij", "olmkn", "trpqs", "xywuv"]
const grid2 = ["abc", "lmp", "qrt"]
console.log(gridChallenge(grid1));
console.log(gridChallenge(grid2));

File: HackerRank/Python/Grid Challenge.py
def gridChallenge(grid):
    # Write your code here

    sorted_grid = []
    for row in grid:
        sorted_grid.append(sorted(row))

    for col in range(len(sorted_grid[0])):
        for row in range(1, len(sorted_grid)):
            if sorted_grid[row - 1][col] > sorted_grid[row][col]:
                return "NO"

    return "YES"


# grid = ["ebacd", "fghij", "olmkn", "trpqs", "xywuv"]
# grid = ["ebacd", "fghij"]


def gridChallenge(grid):
    # Write your code here
    sorted_grid = []
    n = len(grid)  # num of rows
    m = len(grid[0])  # length of col)

    for row in grid:
        sorted_grid.append(sorted(row))

    for row in range(n - 1):
        for col in range(m):
            if sorted_grid[row][col] > sorted_grid[row + 1][col]:
                return "NO"
    return "YES"


grid = ["ebacd", "fghij", "olmkn", "trpqs", "xywuv"]  # "YES"
grid = ["abc", "lmp", "qrt"]  # "YES"
print(gridChallenge(grid))
